che-mount/entrypoint.sh

```diff
@@ -93,45 +113,105 @@ trap 'stop_sync' SIGHUP SIGTERM SIGINT
 init_logging
 init_global_variables
 parse_command_line "$@"
-status=$?
-if [ $status -ne 0 ]; then
-    exit 1
+[ "$(ls -A /mnthost )" ] && EMPTY=false || EMPTY=true
+if [ $EMPTY = false ]; then
+    warn "(che mount): Local folder is not empty. This could overwrite folders/files on local desktop machine or remote workspace machine. Are you sure you want to continue?"
+    read yn
+    case $yn in
+        [Yy]* ) break;;
+        [Nn]* ) exit;;
+        * ) info "(che mount): Please answer yes or no.";;
+    esac
 fi
-
-docker run --rm  -v /var/run/docker.sock:/var/run/docker.sock eclipse/che-action:${CHE_VERSION} get-ssh-data ${WORKSPACE_NAME} ${COMMAND_EXTRA_ARGS} > $HOME/env
+docker run --rm  -v /var/run/docker.sock:/var/run/docker.sock $REPO/che-action:${CHE_VERSION} get-ssh-data ${WORKSPACE_NAME} ${COMMAND_EXTRA_ARGS} > $HOME/env
 if [ $? -ne 0 ]; then
     error "ERROR: Error when trying to get workspace data for workspace named ${WORKSPACE_NAME}"
     echo "List of workspaces are:"
-    docker run --rm  -v /var/run/docker.sock:/var/run/docker.sock eclipse/che-action:${CHE_VERSION} list-workspaces
+    docker run --rm  -v /var/run/docker.sock:/var/run/docker.sock $REPO/che-action:${CHE_VERSION} list-workspaces ${COMMAND_EXTRA_ARGS}
     return 1
 fi
-
 source $HOME/env
 
 # store private key
-mkdir $HOME/.ssh
+mkdir -p $HOME/.ssh
 echo "${SSH_PRIVATE_KEY}" > $HOME/.ssh/id_rsa
-chmod 600 $HOME/.ssh/id_rsa
-
-info "INFO: (che mount): Mounting ${SSH_USER}@${SSH_IP}:/projects with SSHFS"
-sshfs ${SSH_USER}@${SSH_IP}:/projects /mntssh -p ${SSH_PORT} -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no
-
-status=$?
-if [ $status -ne 0 ]; then
-    error "ERROR: Fatal error occurred ($status)"
-    exit 1
-fi
-info "INFO: (che mount): Successfully mounted ${SSH_USER}@${SSH_IP}:/projects (${SSH_PORT})"
-info "INFO: (che mount): Initial sync...Please wait."
-unison /mntssh /mnthost -batch -fat -silent -auto -prefer=newer -log=false > /dev/null 2>&1
-status=$?
-if [ $status -ne 0 ]; then
-    error "ERROR: Fatal error occurred ($status)"
-    exit 1
+chmod 700 $HOME/.ssh/id_rsa
+if [ ${CHE_SYNC_AGENT} = "true" ]; then
```

Comment:
Should I replace with an operand `--sync-agent`? I will need some way for che-mount to determine which one to use... Well actually I could check the workspace machine for unison agent and use it instead of sshfs when available.